src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -338,17 +347,19 @@ private class ListenerFactory : IListenerFactory
             private readonly FunctionDescriptor _descriptor;
             private readonly ITriggeredFunctionExecutor _executor;
             private readonly ITriggerBinding _binding;
+            private readonly IWebJobsExceptionHandler _handler;
 
-            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding)
+            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding, IWebJobsExceptionHandler handler)
             {
                 _descriptor = descriptor;
                 _executor = executor;
                 _binding = binding;
+                _handler = new FunctionExceptionHandler(descriptor, handler);
```

Comment:
wrap the existing handler in a function exception handler, which wraps exceptions in `FunctionInvocationExceptions` so that they are tied to a method and can be picked up by script